### PR TITLE
Feature/auto refresh whitelist

### PR DESCRIPTION
**Pull request by xrdavies**
Add feature to auto-refresh whitelist.
Clean xcode project for fresh guy.

### CODE REVIEW

File: client/http/url.c [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		if ((*str >= '0' && *str <= '9') || *str == '.')
+			str++;
+		else
+			return 0;
+	}
+	return 1;
+}
+
+void parse_query(url_field_t *url, char *query);
+void parse_query(url_field_t *url, char *query)
```
Comment (jonano614) [inline client/http/url.c]: You declare function and define it immediately. Why do you do it?

**Comment by jonano614** [comment]
We already user openssl library as a dependancy. Why did you decide to include whole library to the project?

File: client/http/url.c [hunk 2]
```diff
@@ -0,0 +1,284 @@
+#include "url.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+static char *str_hosttype[] = { "host ipv4", "host ipv6", "host domain", NULL };
+
+//static char *strndup(const char *str, int n)
```
Comment (jonano614) [inline client/http/url.c]: Is there reason to leave this commented function?

File: client/netdb.c [hunk 3]
```diff
@@ -22,6 +23,12 @@
 #define DATABASE            (g_xdag_testnet ? "netdb-testnet.txt" : "netdb.txt")
 #define DATABASEWHITE       (g_xdag_testnet ? "netdb-white-testnet.txt" : "netdb-white.txt")
 
+#define whitelist_url            "https://raw.githubusercontent.com/XDagger/xdag/master/client/netdb-white.txt"
+#define whitelist_url_testnet    "https://raw.githubusercontent.com/XDagger/xdag/master/client/netdb-white-testnet.txt"
+#define WHITE_URL                (g_xdag_testnet ? whitelist_url_testnet : whitelist_url)
```
Comment (jonano614) [inline client/netdb.c]: Can you implement the same for netdb.txt and netdb-testnet.txt? You can do it as a separate branch.

File: client/netdb.c [hunk 4]
```diff
@@ -266,6 +275,41 @@ static void *monitor_thread(void *arg)
...
+	}
+	
+	for (;;) {
+		time_t t = time(0);
+		
+		xdag_mess("try to refresh white-list...");
+		
+		char *resp = http_get(WHITE_URL);
+		if(!resp) {
+			goto nextloop;
```
Comment (jonano614) [inline client/netdb.c]: Is it possible to avoid goto?

File: client/netdb.c [hunk 5]
```diff
@@ -266,6 +275,41 @@ static void *monitor_thread(void *arg)
...
+			fwrite(resp, 1, strlen(resp), f);
+			fclose(f);
+		}
+		pthread_mutex_unlock(&g_white_list_mutex);
+		
+		xdag_info(resp);
+		free(resp);
+		
+nextloop:
+		while (time(0) - t < 67) {
```
Comment (jonano614) [inline client/netdb.c]: Each 67 seconds?

File: client/http/url.c (hunk 2 above)
Comment (xrdavies) [reply client/http/url.c]: It's issue from url.c Not sure what is the purpose of this duplicate function yet. So just leave it here till I figure out what is the purpose.

File: client/netdb.c (hunk 3 above)
Comment (xrdavies) [reply client/netdb.c]: OK, no problem, I will add netdb.txt and netdb-testnet.txt.

File: client/netdb.c (hunk 5 above)
Comment (xrdavies) [reply client/netdb.c]: Yep, just keep the same interval as read white-list.

File: client/http/url.c (hunk 1 above)
Comment (jonano614) [reply client/http/url.c]: If this function is used only in that file you can use static modifier.

File: client/netdb.c (hunk 3 above)
Comment (jonano614) [reply client/netdb.c]: netdb.txt is automatically propagated through network, so it makes sense to find and turn off this logic.

File: client/netdb.c (hunk 4 above)
Comment (jonano614) [reply client/netdb.c]: Better yes.

File: client/netdb.c (hunk 5 above)
Comment (xrdavies) [reply client/netdb.c]: Better to use 15 minutes, 67 seconds is too short.

File: client/netdb.c (hunk 3 above)
Comment (xrdavies) [reply client/netdb.c]: @jonano614 
Checked the code of netdb.txt, it's updated through network. If there is invalid pool not in white-list connect to a valid pool, the valid pool used to white-list this invalid pool, the invalid pool will be in netdb.txt even it is removed from white-list... That's strange..... I will check in another branch.

File: client/http/url.c (hunk 1 above)
Comment (xrdavies) [reply client/http/url.c]: Move the declare above rather than use static since static modifier has different use case.

File: client/netdb.c (hunk 5 above)
Comment (jonano614) [reply client/netdb.c]: Agree, I was thinking to suggest even one hour, but I decided it is just my opinion.

**Comment by jonano614** [comment]
Thank you for your great efforts. I will update win-version.

**Comment by wwens** [comment]
So,  xdag network is running based on whitelist nodes, not a p2p network, right? And, What's the future xdag network is? How to handle 1w nodes?

**Pull request merged**